src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/NodeStatsFixedShardsMetricsCollectorTests.java

```diff
@@ -15,64 +15,120 @@
 
-        String fetchedValue = PerformanceAnalyzerMetrics.getMetric(
-            PluginSettings.instance().getMetricsLocation()
-                + PerformanceAnalyzerMetrics.getTimeInterval(startTimeInMills)+"/indices/NodesStatsIndex/55/");
-        PerformanceAnalyzerMetrics.removeMetrics(PluginSettings.instance().getMetricsLocation()
-            + PerformanceAnalyzerMetrics.getTimeInterval(startTimeInMills));
-        assertEquals("89123.23", fetchedValue);
+        MetricsConfiguration.CONFIG_MAP.put(NodeStatsAllShardsMetricsCollector.class, MetricsConfiguration.cdefault);
+        collector = new NodeStatsFixedShardsMetricsCollector(controller);
+    }
 
+    @After
+    public void tearDown() throws Exception {
+        super.tearDown();
+    }
 
+    @Test
+    public void testNodeStatsMetrics() {
         try {
-            nodeStatsFixedShardsMetricsCollector.saveMetricValues("89123.23", startTimeInMills, "NodesStatsIndex");
+            collector.saveMetricValues("89123.23", startTimeInMills, "NodesStatsIndex");
             assertTrue("Negative scenario test: Should have been a RuntimeException", true);
         } catch (RuntimeException ex) {
             //- expecting exception...only 1 values passed; 2 expected
         }
 
         try {
-            nodeStatsFixedShardsMetricsCollector.saveMetricValues("89123.23", startTimeInMills);
+            collector.saveMetricValues("89123.23", startTimeInMills);
             assertTrue("Negative scenario test: Should have been a RuntimeException", true);
         } catch (RuntimeException ex) {
             //- expecting exception...only 0 values passed; 2 expected
         }
 
         try {
-            nodeStatsFixedShardsMetricsCollector.saveMetricValues("89123.23", startTimeInMills, "NodesStatsIndex", "55", "123");
+            collector.saveMetricValues("89123.23", startTimeInMills, "NodesStatsIndex", "55", "123");
             assertTrue("Negative scenario test: Should have been a RuntimeException", true);
         } catch (RuntimeException ex) {
             //- expecting exception...only 3 values passed; 2 expected
         }
 
         try {
-            nodeStatsFixedShardsMetricsCollector.getNodeIndicesStatsByShardField();
+            collector.getNodeIndicesStatsByShardField();
         } catch (Exception exception) {
             assertTrue("There shouldn't be any exception in the code; Please check the reflection code for any changes", true);
         }
 
+        collector = new NodeStatsFixedShardsMetricsCollector(null);
+        try {
+            collector.collectMetrics(startTimeInMills);
+        } catch (Exception exception) {
+            assertTrue("There shouldn't be any exception in the code; Please check the reflection code for any changes", true);
+        }
+    }
+
+    @Test
+    public void testCollectMetrics() {
+        createIndex(TEST_INDEX);
+        Mockito.when(controller.getNodeStatsShardsPerCollection()).thenReturn(1);
```

Comment:
Should we add a test to check if all the metrics for all the shards are populated in subsequent runs of this collector?